spirelayout.cpp

```diff
@@ -153,7 +153,7 @@ bool TrapUpgrades::operator<(const TrapUpgrades &other) const
 string TrapUpgrades::str() const
 {
 	char buf[4];
-	buf[0] = '0'+fire;
+	buf[0] = fire > 9 ? 'A'+fire-10 : '0'+fire;
```

Comment:
pretty janky but I didn't find a clean way to go from a number to just a character